aws/src/main/java/org/apache/iceberg/aws/s3/BaseS3File.java

```diff
@@ -77,6 +78,12 @@ public boolean exists() {
 
   protected HeadObjectResponse getObjectMetadata() throws S3Exception {
     if (metadata == null) {
+      // list object to force strong consistency
+      client().listObjectsV2(ListObjectsV2Request.builder()
+          .bucket(uri().bucket())
+          .prefix(uri().key())
```

Comment:
you might want to include `maxKeys(1)` as well since we only care about a single entry.

Reply:
Do we want to validate, or just set metadata to null so that `exists()` return false? I understand that it is not likely that a file name is a prefix of another file, but we cannot eliminate the possibility of some location provider with such behavior.

Reply:
See the updated logic for more details